Sources/MongoKitten/ChangeStream.swift

```diff
@@ -139,25 +151,45 @@ public struct ChangeStream<T: Decodable> {
     }
     
     /// Iterates over the change stream notifications and calls the given handler for each notification
+    /// If the handler throws an error, or when a network error occurs, the task will be failed with that error
+    /// 
     /// - Parameter handler: The handler to call for each notification
     /// - Returns: A task that will be completed when the change stream is drained. Can be cancelled to stop the change stream
-    /// - Throws: If the handler throws an error, the task will be failed with that error
     @discardableResult
+    @available(*, deprecated, message: "Use `forEach` returning `ChangeStreamPollingResult` instead")
     public func forEach(handler: @escaping @Sendable (Notification) async throws -> Bool) -> Task<Void, Error> {
         Task {
-            while !cursor.isDrained {
-                for element in try await cursor.nextBatch() {
-                    if try await !handler(element) {
-                        return
-                    }
-                }
-                
-                if let getMoreInterval = self.getMoreInterval {
-                    try await Task.sleep(nanoseconds: UInt64(getMoreInterval.nanoseconds))
+            try await forEach { notification in
+                if try await handler(notification) {
+                    return .continue
                 } else {
-                    try Task.checkCancellation()
+                    return .stop
```

Comment:
Isn't this a breaking change if somebody relies on catching cancellation errors?

Reply:
This is still the case:

https://github.com/orlandos-nl/MongoKitten/pull/329/files#diff-120634bc6a29b714a4b68a7f15f5a4fe6a6ef678bbf0aa5abeafef9eed7ef0e2R191

Reply:
Right, the diff was a bit confusing. Thanks for clarifying 🙃